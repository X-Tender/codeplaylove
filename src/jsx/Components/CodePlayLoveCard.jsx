import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
	title: PropTypes.string.isRequired,
	children: PropTypes.node.isRequired,
	style: PropTypes.string.isRequired,
};

export const CodePlayLoveCard = ({ title, children, style }) => (
	<li className={`code-play-love-card code-play-love-card--${style}`}>
		<div className={`code-play-love-card__icon code-play-love-card__icon--${style}`} />
		<div className="code-play-love-card__content">
			<h3 className="code-play-love-card__head">{title}</h3>
			{children}
		</div>
	</li>
);

CodePlayLoveCard.propTypes = propTypes;

export default CodePlayLoveCard;
