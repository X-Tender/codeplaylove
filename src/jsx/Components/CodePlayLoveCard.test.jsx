import React from 'react';
import { shallow } from 'enzyme';
import CodePlayLoveCard from './CodePlayLoveCard';

describe('CodePlayLoveCard', () => {
	const style = 'code';
	const title = 'Code';
	const children = 'This is code';

	const comp = shallow(
		<CodePlayLoveCard style={style} title={title}>
			{children}
		</CodePlayLoveCard>
	);

	it('renders correctly', () => {
		expect(comp).toMatchSnapshot();
	});

	it('renders the correct props', () => {
		expect(comp.find(`li.code-play-love-card.code-play-love-card--${style}`).exists()).toBe(true);
		expect(comp.find('h3.code-play-love-card__head').text()).toEqual(title);
		expect(comp.find('div.code-play-love-card__content').text()).toEqual(title + children);
	});
});
