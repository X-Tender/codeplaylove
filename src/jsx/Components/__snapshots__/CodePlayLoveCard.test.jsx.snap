// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodePlayLoveCard renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CodePlayLoveCard
    style="code"
    title="Code"
  >
    This is code
  </CodePlayLoveCard>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="code-play-love-card__icon code-play-love-card__icon--code"
        />,
        <div
          className="code-play-love-card__content"
        >
          <h3
            className="code-play-love-card__head"
          >
            Code
          </h3>
          This is code
        </div>,
      ],
      "className": "code-play-love-card code-play-love-card--code",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "code-play-love-card__icon code-play-love-card__icon--code",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="code-play-love-card__head"
            >
              Code
            </h3>,
            "This is code",
          ],
          "className": "code-play-love-card__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Code",
              "className": "code-play-love-card__head",
            },
            "ref": null,
            "rendered": "Code",
            "type": "h3",
          },
          "This is code",
        ],
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="code-play-love-card__icon code-play-love-card__icon--code"
          />,
          <div
            className="code-play-love-card__content"
          >
            <h3
              className="code-play-love-card__head"
            >
              Code
            </h3>
            This is code
          </div>,
        ],
        "className": "code-play-love-card code-play-love-card--code",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "code-play-love-card__icon code-play-love-card__icon--code",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="code-play-love-card__head"
              >
                Code
              </h3>,
              "This is code",
            ],
            "className": "code-play-love-card__content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Code",
                "className": "code-play-love-card__head",
              },
              "ref": null,
              "rendered": "Code",
              "type": "h3",
            },
            "This is code",
          ],
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
